crates/router/src/connector/checkout/transformers.rs

```diff
@@ -551,3 +552,109 @@ impl From<CheckoutRedirectResponseStatus> for enums::AttemptStatus {
         }
     }
 }
+
+pub fn is_refund_event(event_code: &CheckoutTxnType) -> bool {
+    matches!(
+        event_code,
+        CheckoutTxnType::PaymentRefunded | CheckoutTxnType::PaymentRefundDeclined
+    )
+}
+
+pub fn is_chargeback_event(event_code: &CheckoutTxnType) -> bool {
+    matches!(
+        event_code,
+        CheckoutTxnType::DisputeReceived
+            | CheckoutTxnType::DisputeExpired
+            | CheckoutTxnType::DisputeAccepted
+            | CheckoutTxnType::DisputeCanceled
+            | CheckoutTxnType::DisputeEvidenceSubmitted
+            | CheckoutTxnType::DisputeEvidenceAcknowledgedByScheme
+            | CheckoutTxnType::DisputeEvidenceRequired
+            | CheckoutTxnType::DisputeArbitrationLost
+            | CheckoutTxnType::DisputeArbitrationWon
+            | CheckoutTxnType::DisputeWon
+            | CheckoutTxnType::DisputeLost
+    )
+}
+
+#[derive(Debug, Deserialize)]
+pub struct CheckoutWebhookData {
+    pub id: String,
+    pub payment_id: Option<String>,
+    pub action_id: Option<String>,
+    pub amount: i32,
+    pub currency: String,
+    #[serde(default, with = "common_utils::custom_serde::iso8601::option")]
+    pub evidence_required_by: Option<PrimitiveDateTime>,
+    pub reason_code: Option<String>,
+    #[serde(default, with = "common_utils::custom_serde::iso8601::option")]
+    pub date: Option<PrimitiveDateTime>,
+}
+#[derive(Debug, Deserialize)]
+pub struct CheckoutWebhookBody {
+    #[serde(rename = "type")]
+    pub txn_type: CheckoutTxnType,
+    pub data: CheckoutWebhookData,
+    #[serde(default, with = "common_utils::custom_serde::iso8601::option")]
+    pub created_on: Option<PrimitiveDateTime>,
+}
+#[derive(Debug, Deserialize, strum::Display, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum CheckoutTxnType {
```

Comment:
Would be great if we could call this `CheckoutTransactionType` instead. Same for the field in `CheckoutWebhookBody`, please change it to `transaction_type`. We avoid abbreviations like `transaction -> txn` as much as possible in our code, and use verbose names instead.